src/components/MoneyRequestHeader.tsx

```diff
@@ -171,8 +173,11 @@ function MoneyRequestHeader({report, parentReportAction, policy, shouldUseNarrow
     }
 
     useEffect(() => {
-        setShouldShowHoldMenu(isOnHold && !shownHoldUseExplanation);
-    }, [isOnHold, shownHoldUseExplanation]);
+        if (isLoadingHoldUseExplained) {
+            return;
+        }
+        setShouldShowHoldMenu(isOnHold && !holdUseExplained);
```

Comment:
I have found the code that causes the issue and pushed a new fix. We can't set the prop default value as the onyx initial value because it will be used when the data is still loading by the onyx. Instead, I do an early return when the data is still loading.